matrixprofile/matrixProfile.py

```diff
@@ -52,6 +103,9 @@ def _matrixProfile(tsA,m,orderClass,distanceProfileFunction,tsB=None):
     order = orderClass(len(tsA)-m+1)
     mp, mpIndex = _self_join_or_not_preprocess(tsA, tsB, m)
 
+    tsA = _clean_nan_inf(tsA)
```

Comment:
All of these checks will fail in the case of tsB being None. You need to check for a self join first and copy tsA to tsB. After that you can call _clean_nan_inf(tsA) etc.

```python
if not is_array_like(tsB):
    tsB = tsA
```